dapp/src/components/TransactionModal/ActiveStakingStep/DepositBTCModal.tsx

```diff
@@ -63,8 +69,16 @@ export default function DepositBTCModal() {
       } else {
         onError(error)
       }
+
+      const captureParameters =
+        error instanceof Error
+          ? {
+              cause: error.message,
+            }
+          : undefined
+      handleCapture(PostHogEvent.DepositFailure, captureParameters)
```

Comment:
Maybe we should extract this code to a separate hook or util function? We do the same in `SignMessageModal.tsx` file.

Reply:
Nice one
Ref commit: fa12d8a